massa-pool-worker/src/endorsement_pool.rs

```diff
@@ -115,6 +129,12 @@ impl EndorsementPool {
                         panic!("endorsement is expected to be absent from endorsements_sorted at this point");
                     }
                     added.insert(endo.id);
+
+                    // Send endorsements to Denunciation Factory
+                    let de_interest = DenunciationInterest::WrappedEndorsement(endo.clone());
```

Comment:
The operations and header are sent to denunciation_factory from protocol why endorsement is sent from the pool ?

Reply:
If I am not mistaken, the endorsements from both the network and the endorsement factory are stored in the endorsement pool and thus we can feed the denunciation factory from only one place. What do you think?

Reply:
It's the same for block header that can be produce from block factory they are both sent to consensus. Consensus receive them all I think it should be better to make this send from the consensus.

Reply:
I think it makes more sense that it comes from the pools but your point stands too. What do you think @damip ?